lazyflow/slot.py

```diff
@@ -1102,13 +1102,13 @@ def setValue(self, value, notify=True, check_changed=True, extra_meta={}):
                     same = value is self._value
                     if not same:
                         try:
-                            same = value == self._value
+                            same = numpy.array_equal(value, self._value)
                         except ValueError:
                             # Some values can't be compared with __eq__,
                             # in which case we assume the values are different
                             same = False
-                        if isinstance(same, (numpy.ndarray, TinyVector)):
-                            same = same.all()
+
+                        assert isinstance(same, bool)
```

Comment:
I can't see any path where `type(same) != bool`... is this still necessary?

Reply:
me neither, but I am paranoid :ghost: